src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -114,12 +196,30 @@ public unsafe void Destroy()
 
             private unsafe IntPtr AsRuntimeDefined(in Guid riid)
             {
-                if ((Flags & CreateComInterfaceFlags.CallerDefinedIUnknown) == CreateComInterfaceFlags.None)
+                if ((Flags & CreateComInterfaceFlagsEx.CallerDefinedIUnknown) == CreateComInterfaceFlagsEx.None)
                 {
                     if (riid == IID_IUnknown)
                     {
                         return (IntPtr)(Dispatches + UserDefinedCount);
                     }
+
+                    if ((Flags & CreateComInterfaceFlagsEx.TrackerSupport) == CreateComInterfaceFlagsEx.TrackerSupport)
+                    {
+                        if (riid == IID_IReferenceTrackerTarget)
+                        {
+                            return (IntPtr)(Dispatches + UserDefinedCount + 1);
+                        }
+                    }
+                }
+                else
+                {
+                    if ((Flags & CreateComInterfaceFlagsEx.TrackerSupport) == CreateComInterfaceFlagsEx.TrackerSupport)
```

Comment:
It would be nice to avoid this duplication by maintaining a running index:

```
int i = UsedDefinedCount;

if ((Flags & CreateComInterfaceFlagsEx.CallerDefinedIUnknown) == CreateComInterfaceFlagsEx.None)
{
    if (riid == IID_IUnknown)
    {
        return (IntPtr)(Dispatches + i);
    }
    i++;
}

if ((Flags & CreateComInterfaceFlagsEx.TrackerSupport) == CreateComInterfaceFlagsEx.TrackerSupport)
{
....
```